### ISSUE TITLE
UX: offline_kube_version via CLI, reduce confusion

### ISSUE THREAD

**Issue by needleshaped** [helmwave/helmwave#1037]
**Describe the solution you'd like**
---

TLDR: global CLI flag for `offline_kube_version` (overriden by YAML), for example:
  - `helmwave build --offline-kube-version=1.26.11` which MUST require `--diff-mode=[ none | local ]` (and vice versa).

P.S. maybe a more linear (.. logically named/grouped?) configuration options may be better as final solution, (like `--diff-mode=` + `--kube-version=[ online | SEMVER ]` + renaming `offline_kube_version` to just `kube_version` or very precise `kube_version_offline_fallback`), but that's too big to ask.

In this case PR to docs made below would not be necessary, as options themselves would force you to use them in conjunction.

**Is your feature request related to a problem? Please describe.**
---

Seems offline Kubernetes scenario is source of regular confusion among newcomers (me included). Example issues: #886 #651 #537.

It costs 5-15 minutes to find out how to achieve it.

- `helmwave build --help` says little
    - _made PR to improve user experience for current version: https://github.com/helmwave/helmwave/pull/1036 ✅_
- Documentation in this case is more developer, than user-oriented.
    - _Ok, Made PR to documentation about [build](https://docs.helmwave.app/latest/cli/#build) and [diff-mode](https://docs.helmwave.app/latest/cli/#-diff-mode) improve user experience for current version: https://github.com/helmwave/docs/pull/95 ✅_

But even when configured, it does not feel *logical* for simple use case.
- why combination of CLI with YAML? (ok, answer is: one for diff, other to get version for schema validation)
- why `offline_kube_version` is per-release, not global? (ok, answer is: [because release can have custom context](https://docs.helmwave.app/latest/yaml/#context))
- why `offline_kube_version` is YAML, not CLI option? (ok, answer is same as above, so need possibility to define different values for each release)

So, there are answers, and now we can ask other question:
- why most common simple use case (single cluster but temporary offline) is forced to be **complex**
- why for this use case we need to define `offline_kube_version`?
    - After local development actions (goal: get template, check `helmwave.yml.tpl` is correct outside of CI/CD) these settings become obsolete, and once again require step to add, step to remove.
    - It increases **maintenance** cost, I don's see now who would want to have/support in config, as versions are bound to change.

**Describe alternatives you've considered**
---
Improved documentation in https://github.com/helmwave/docs/pull/95 and https://github.com/helmwave/helmwave/pull/1036

**Additional context**
---
P.S. Thanks for interesting project. Coming from Helm / Helm Library / Helmfile, I'm learning new tool, but must admit, learning curve is steeper, then I expected. Helmfile felt more.. natural. So, I'm trying to help)

**Comment by zhilyaev** [comment]
# Hi @needleshaped ✋🏼 , thanks for your contributions! It has a strong point and is so detailed 🤝 

I just would like to keep my explanations about some things here.

I am convinced that only a community can write the best documentation. When you write code, it's difficult to provide clear features for customers especially if your product is a top-notch tool (It's about Documentation in this case is more developer, than user-oriented). Each feature was introduced at a different time. Thus so, you can see that offline_kube_version expected diff-mode. I see we should improve our processes when features affect each other. 

## Scoping for release config parameters

`helmwave.yml` is the main file with all releases in a stack.
Diff doesn't affect rendered manifests and planfile, **it's just UI.**
Offline_kube_version will change behavior to build a plandir. 
Actually, Offline_kube_version doesn't need a kube context. 

## Why not force diff-mode=local if Offline_kube_version was set up?
_TLDR: CRD._
When you want to install CRD first. Your Kube API server can reject your CRD. 

Well, It is possible to use `diff mode=local` and `offline_kube_version` as **independent** parameters.
More often than not everyone needs true offline mode,  It means you have to enable both of them.

1. `diff-mode=local`– I just want to compare with your local dir. `helmwave build --diff-mode=local --plandir=my-prev-plan` It's very useful, for debugging before applying. 
2. `diff-mode=live` – I want to compare with manifests in the cluster. 
3. `offline_kube_version=1.27.1` – I don't want to get the kube version from an online cluster. For example: installation CRD. 
4. `diff-mode=live && offline_kube_version=1.27.1` – I want to compare with the online Kubernetes cluster. But I force the kube-version for CRD.
5. `diff-mode=local && offline_kube_version=1.27.1` – it's true offline mode. Don't ask Kube API at all.  


## I left my thoughts about a related problem here. 

Sometimes, I found the issues. When contributors want to pass some release config parameters via CLI.
I've suggested the next feature request: Allow passing any parameters to the default release config via CLI or ENV. 

## Allow passing any parameters to the default release config via CLI or ENV. 

`cat helmwave.yml`

```yaml
releases:
 - name: my-app
```

```bash
export HELMWAVE_RELEASE_OFFLINE_KUBE_VERSION=1.27.1
export HELMWAVE_RELEASE_TIMEOUT=10m
export HELMWAVE_RELEASE_CHART_NAME=my/chart
```

`helmwave build --release-timout=10 --chart-name=my/chart --offline_kube_version=1.21.1`



It will be transformed to 

```yaml
releases:
 - name: my-app
    offline_kube_version: 1.27.1
    timeout: 10m
    chart:
      name: my/chart
```

I see benefits. That's keep simple.

**limitations:**
- I need some time to investigate how to deal with uniqname parameters like name, namespace, context here.
- I would like to save helmwave.yml like the main file with all releases in a stack.


### But you can use the next `helmwave.yml.tpl` for these cases right now!


```yaml
releases:
 - name: my-app
    offline_kube_version: {{ env "HELMWAVE_RELEASE_OFFLINE_KUBE_VERSION" }}
    timeout: {{ env "HELMWAVE_RELEASE_TIMEOUT" }}
    chart:
      name: {{ env "HELMWAVE_RELEASE_CHART_NAME" }}
```

**Comment by zhilyaev** [comment]
@needleshaped feel free to discuss in our telegram community [https://t.me/helmwave](https://t.me/+WHt1epQ9NvN-3R8c)
